internal/runner/options.go

```diff
@@ -0,0 +1,104 @@
+package runner
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/projectdiscovery/fileutil"
+	"github.com/projectdiscovery/goflags"
+	"github.com/projectdiscovery/gologger"
+)
+
+type Output struct {
+	Timestamp time.Time `json:"timestamp,omitempty"`
+	IP        string    `json:"ip"`
+	Cdn       bool      `json:"cdn,omitempty"`
+	CdnName   string    `json:"cdn_name,omitempty"`
+	itemType  string
+}
+
+func (o *Output) String() string {
+	return fmt.Sprintf("%s [%s] [%s]", o.IP, o.itemType, o.CdnName)
+}
+func (o *Output) StringIP() string {
+	return o.IP
+}
+
+type Options struct {
+	inputs      goflags.StringSlice
+	list        string
+	response    bool
+	hasStdin    bool
+	output      string
+	version     bool
+	json        bool
+	cdn         bool
+	cloud       bool
+	waf         bool
+	exclude     bool
+	matchCdn    goflags.StringSlice
+	matchCloud  goflags.StringSlice
+	matchWaf    goflags.StringSlice
+	filterCdn   goflags.StringSlice
+	filterCloud goflags.StringSlice
+	filterWaf   goflags.StringSlice
+}
+
+func ParseOptions() *Options {
+	opts, err := readFlags()
+	if err != nil {
+		gologger.Fatal().Msgf("Program Exiting: %s\n", err)
+	}
+	return opts
+}
+
+// readFlags reads the flags and options for the utility
+func readFlags() (*Options, error) {
+	opts := &Options{}
+	opts.hasStdin = fileutil.HasStdin()
+
+	flagSet := goflags.NewFlagSet()
+	flagSet.SetDescription("cdncheck is a tool for identifying the technology associated with ip network addresses.")
+
+	flagSet.CreateGroup("input", "Input",
+		flagSet.StringSliceVarP(&opts.inputs, "inputs", "i", nil, "inputs to process", goflags.CommaSeparatedStringSliceOptions),
+		flagSet.StringVarP(&opts.list, "list", "l", "", "file with inputs to process"),
+	)
+
+	flagSet.CreateGroup("detection", "Detection",
+		flagSet.BoolVarP(&opts.cdn, "cdn", "", false, "display cdn ip in cli output"),
+		flagSet.BoolVarP(&opts.cloud, "cloud", "", false, "display cloud ip in cli output"),
+		flagSet.BoolVarP(&opts.waf, "waf", "", false, "display waf ip in cli output"),
+	)
+
+	flagSet.CreateGroup("output", "Output",
+		flagSet.BoolVarP(&opts.response, "resp", "", false, "display technology name in cli output"),
+		flagSet.StringVarP(&opts.output, "output", "o", "", "write output in plain format to file"),
+		flagSet.BoolVarP(&opts.version, "version", "", false, "display version of the project"),
+		flagSet.BoolVarP(&opts.json, "json", "", false, "write output in json format to file"),
+	)
+
+	flagSet.CreateGroup("matchers", "Matchers",
+		flagSet.StringSliceVarP(&opts.matchCdn, "match-cdn", "mcdn", nil, "match host with specified cdn provider (fastly, incapsula)", goflags.CommaSeparatedStringSliceOptions),
```

Comment:
These values need to be dynamic, similar to https://github.com/projectdiscovery/httpx/blob/f2cd5b835348c5563c4c018dc2ed414d82ccf249/runner/options.go#L309, currently `fastly, incapsula` is hard coded for all options.

Reply:
Done